### DISCUSSION TITLE
How to show parent_need (from list2need) as column in needtable supported?

### DISCUSSION THREAD

**Discussion by daniel3097** [useblocks/sphinx-needs#876] [Q&A]
list2needs seems to create links of types parent_need and child_need.
but the related columns in needtable are always empty.

This is what i am doing:
```rst
.. list2need::
   :types: test, run, issue
   :presentation: standalone
   :delimiter: .

   * (SYE_T00) Test . ((links="SYE_R06"))
     * (SYE_TR00) Test Run. ((links="SYE_R06"))
       * (SYE_I00) Issue. ((links="SYE_R06"))


.. needtable:: Test Run
   :style: datatable
   :types: run
   :columns: id;title;status;parent_need;links_back;links
```

Is this use case not supported, a typo or did  i miss something from the documentation? 
Is it because parent_need is  an external link and behaves differently than other link types?

**Comment by danwos** [comment]
I think it may be also  ``parent_needs`` and ``parent_needs_back``, so with an ending ``s``.
At least for the links.

``parent_need`` does also exist, but just as string value.
But this should also work. I'm confused why not.

I just made a quick check in one of my existing docs, and it works: list2need + needtable with `parent_need`.

So yes, it is supported and should work.
Can you create a small working sphinx-project, which shows this error?
Then I'm able to dive much deeper into the problem.

**Reply by daniel3097** [reply]
thank you, it did not solve the issue - and I will follow your advice (create a small project).

right now I am exploring the tool and I am impressed in general
I also experimented with latexpdf builder - but get many warnings and the visual output rendering of needs is not fully convincing ...

**Reply by danwos** [reply]
Yes, Latex output is not the best and I have not much skills in it.
Therefore I created also Sphinx-SimplePDF, which is based on HTML and CSS only: https://sphinx-simplepdf.readthedocs.io

There the output of Need objects is looking much better :)

**Reply by daniel3097** [reply]
your work on Sphinx-SimplePDF is impressive - way better than the latexpdf path!!
but installation on windows looks complicated and the GTK3 dependency is tricky and might be banned by our IT policy :-(.

I therefore started looking into simplified layout and grid with only content and head ... .

**Reply by danwos** [reply]
Ohh yes, windows support is not so good and I have never tested it.
Most known windows users are using Sphinx-SimplePDF inside WSL2 or inside a docker environment.
Maybe this could be a workaround for you as well.

But if you have Latex-Skills and are able to identify our problems with it, feel free to create some issues or PRs.
I will try to support there as much as I can.

**Reply by daniel3097** [reply]
I have no latex skills and limited python coding skills. But i can read, understand and hack some code.

One idea to improve the sphinx-needs pdf latex output is to support a grid-system with two rows with head/content.
(tsee quick hack in attached pdf page 56):

[sphinx_needs_latexpdf.pdf](https://github.com/useblocks/sphinx-needs/files/10892951/sphinx_needs_latexpdf.pdf)

A second idea is to get rid of the vertical space before and after the non content row. The way i understand the code so far that's happening because of some sphinx or docutils code how to render the literal block to latex.

A third idea is to hack the layout string and allow some kind of f-string. so meta information and head can be rendered on one or two lines.

I will share a demo project when there is something promising

**Reply by danwos** [reply]
Another idea is to register some Latex-specific Sphinx visitor function which handles the problematic docutils-node in a different way as the normal latex builder does.

If we touch the docutils-nodes earlier, I guess it will have an impact on the HTML build as well. 
So we should try to avoid it or have builder-specific flags in areas of the code, which are not part of the builders itself.
